mrp_subcontracting_partner_management/models/res_partner.py

```diff
@@ -0,0 +1,267 @@
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    is_subcontractor_partner = fields.Boolean(string="Subcontractor")
+    subcontracted_created_location_id = fields.Many2one("stock.location")
```

Comment:
Why do you need this extra field instead of simply relying on the standard one that defines the subcontractor location?

Reply:
Because this field always contains a record and we need to find some way to distinguish the default location from the one we have created ourselves. Here is an example of why it is needed for:
1. Enable "Subcontractor". A new location is created.
2. Now disable "Subcontractor". Location is archived.
3. Now let's re-enable "Subcontractor". At this point we need to unarchive an existing location. Otherwise a new location will be created again. This is what this field is used for.

I understand your concerns regarding "too much fields in partner". But we need to store the information somewhere. Another option is to store in the stock.location We can opt for this if you prefer

Reply:
@pedrobaeza what do you think ?

Reply:
You are already adding `is_subcontractor_partner` for distinguishing if the partner has already been setup or not, so no need of these extra fields.

Reply:
> You are already adding `is_subcontractor_partner` for distinguishing if the partner has already been setup or not, so no need of these extra fields.

It distinguishes only the current status of the partner. If we remove these fields then next time when the subcontractor is re-enabled new records will be created instead of unarchiving the existing ones.

Reply:
I understand what you mean, but being a corner case that you play with activating and deactivating the field, I would go with:

- Using standard fields.
- If you unmark the check, you deactivate the records whatever they have linked.
- If you mark (again or not), check if the linked location is active. If not, you activate if. If active, then you create a new one, as it can be deduced that is the standard one which is linked.

What do you think?

Reply:
Yes, you are absolutely right. However in some cases we cannot distinguish the location that was archived by our module from the location that was archived by user explicitly.

Example:
- Archive the default "Subcontracting" location. 
- Install this module.
- Enable the "Subcontractor" for  some partner.

Expected behavior:
- A new location is created "Subcontracting/PartnerName"

Actual behavior without special fields:
- The default "Subcontracting" location will be unarchived, no location will be created for the partner

Reply:
@pedrobaeza looks reasonable? :)

Reply:
I know, but as said, it's a corner case that is not going to happen a lot, and easily fixable. I prefer to not add too much fields, but I don't want to impose my criteria. Go the way you prefer.